pymc/initvalues.py

```diff
@@ -0,0 +1,212 @@
+                return_transformed=True,
+                overrides=convert_start_dict_to_initval_dict(model, chain_start or None),
+            )
+            for chain_start in initvals
+        ]
+    else:
+        raise ValueError(
+            f"Number of start dicts ({len(initvals)}) does not match the number of seeds ({len(seeds)})."
+        )
+
+    # The chain-wise seed is respected for the start point evaluation.
+    start_points = [ipfn(seed) for ipfn, seed in zip(ipfns, seeds)]
+    return start_points
+
+
+def make_initial_point_expression(
+    *,
+    free_rvs: Sequence[TensorVariable],
+    rvs_to_values: Dict[TensorVariable, TensorVariable],
+    initval_strategies: Dict[TensorVariable, Optional[Union[np.ndarray, Variable, str]]],
+    jitter_rvs: Set[TensorVariable] = None,
+    default_strategy: str = "prior",
+    return_transformed: bool = False,
+) -> List[TensorVariable]:
+    """Creates the tensor variables that need to be evaluated to obtain an initial point.
+
+    Parameters
+    ----------
+    free_rvs : list
+        Tensors of free random variables in the model.
+    rvs_to_values : dict
+        Mapping of free random variable tensors to value variable tensors.
+    initval_strategies : dict
+        Mapping of free random variable tensors to initial value strategies.
+        For example the `Model.initial_values` dictionary.
+    jitter_rvs : set
+        The set (or list or tuple) of random variables for which a U(-1, +1) jitter should be
+        added to the initial value. Only available for variables that have a transform or real-valued support.
+    default_strategy : str
+        Which of { "moment", "prior" } to prefer if the initval strategy setting for an RV is None.
+    return_transformed : bool
+        Switches between returning the tensors for untransformed or transformed initial points.
+
+    Returns
+    -------
+    initial_points : list of TensorVariable
+        Aesara expressions for initial values of the free random variables.
+    """
+    from pymc.distributions.distribution import get_moment
+
+    if jitter_rvs is None:
+        jitter_rvs = set()
+
+    initial_values = []
+    initial_values_transformed = []
+
+    for variable in free_rvs:
+        strategy = initval_strategies.get(variable, None)
+
+        if strategy is None:
+            strategy = default_strategy
+
+        if strategy == "moment":
+            value = get_moment(variable)
+        elif strategy == "prior":
+            value = variable
+        else:
+            value = at.as_tensor(strategy, dtype=variable.dtype).astype(variable.dtype)
+
+        transform = getattr(rvs_to_values[variable].tag, "transform", None)
+
+        if transform is not None:
+            value = transform.forward(variable, value)
+
+        if variable in jitter_rvs:
+            jitter = at.random.uniform(at.zeros_like(variable) - 1, at.zeros_like(variable) + 1)
+            jitter.name = "jitter"
```

Comment:
```suggestion
            jitter.name = variable.name + "_jitter"
```

Maybe we could prefix the jitter with the variable's name?